dlp/dlp_snippets/main.go

```diff
@@ -0,0 +1,304 @@
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// dlp is an example of using the DLP API.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/net/context"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"github.com/fatih/color"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+type minLikelihoodFlag struct {
+	l dlppb.Likelihood
+}
+
+func (m *minLikelihoodFlag) String() string {
+	return fmt.Sprint(m.l)
+}
+
+func (m *minLikelihoodFlag) Set(s string) error {
+	l, ok := dlppb.Likelihood_value[s]
+	if !ok {
+		return fmt.Errorf("not a valid likelihood: %q", s)
+	}
+	m.l = dlppb.Likelihood(l)
+	return nil
+}
+
+func minLikelihoodValues() string {
+	var s []string
+	for _, m := range dlppb.Likelihood_name {
+		s = append(s, m)
+	}
+	return strings.Join(s, ", ")
+}
+
+type bytesTypeFlag struct {
+	bt dlppb.ByteContentItem_BytesType
+}
+
+func (f *bytesTypeFlag) String() string {
+	return fmt.Sprint(f.bt)
+}
+
+func (f *bytesTypeFlag) Set(s string) error {
+	b, ok := dlppb.ByteContentItem_BytesType_value[s]
+	if !ok {
+		return fmt.Errorf("not a valid BytesType: %q", s)
+	}
+	f.bt = dlppb.ByteContentItem_BytesType(b)
+	return nil
+}
+
+func bytesTypeValues() string {
+	var s []string
+	for _, m := range dlppb.ByteContentItem_BytesType_name {
+		s = append(s, m)
+	}
+	return strings.Join(s, ", ")
+}
+
+func main() {
+	ctx := context.Background()
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	project := flag.String("project", "", "GCloud project ID (required)")
+	languageCode := flag.String("languageCode", "en-US", "Language code for infoTypes")
+	maxFindings := flag.Int("maxFindings", 0, "Number of results for inspect*, createTrigger, and createInspectTemplate (default 0 (no limit))")
+	includeQuote := flag.Bool("includeQuote", false, "Include a quote of findings for inspect* (default false)")
+	infoTypesString := flag.String("infoTypes", "PHONE_NUMBER,EMAIL_ADDRESS,CREDIT_CARD_NUMBER,US_SOCIAL_SECURITY_NUMBER", "Info types to inspect*, redactImage, createTrigger, and createInspectTemplate")
+
+	var minLikelihood minLikelihoodFlag
+	flag.Var(&minLikelihood, "minLikelihood", fmt.Sprintf("Minimum likelihood value for inspect*, redactImage, createTrigger, and createInspectTemplate [%v] (default %v)", minLikelihoodValues(), dlppb.Likelihood_name[0]))
+
+	var bytesType bytesTypeFlag
+	flag.Var(&bytesType, "bytesType", fmt.Sprintf("Bytes type of input file for inspectFile and redactImage [%v] (default %v)", bytesTypeValues(), dlppb.ByteContentItem_BytesType_name[0]))
```

Comment:
i somewhat prefer these being package-level variables. they get formatted nicely in a block and package wide visibility is useful. (but i guess not useful in this case)

Reply:
Done.